crates/runtime/src/memory.rs

```diff
@@ -315,6 +315,15 @@ impl Memory {
     ) -> Result<Self> {
         let (minimum, maximum) = Self::limit_new(plan, store)?;
 
+        if base.len() < minimum {
+            bail!(
+                "memory allocation of {} bytes does not meet this module's \
```

Comment:
Another nit:

This error message is a little confusing to me as it reads like the requirement is in the module and not that the module's initial memory size exceeded the maximum configured for the pooling allocator.

Perhaps `initial memory size of {} bytes exceeds the pooling allocator's configured maximum memory size of {} bytes` or something along those lines?